### PR TITLE
chore: Add OnAcknowledgmentPacket to IBCModule V2 Interface

### PR DESCRIPTION
**Pull request by bznein**
## Description

Somewhat forgot about this one!

closes: #7410

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: testing/mock/v2/ibc_app.go [hunk 1]
```diff
@@ -9,7 +9,8 @@ import (
 )
 
 type IBCApp struct {
-	OnSendPacket    func(ctx context.Context, sourceID string, destinationID string, sequence uint64, data channeltypesv2.PacketData, signer sdk.AccAddress) error
-	OnRecvPacket    func(ctx context.Context, sourceID string, destinationID string, data channeltypesv2.PacketData, relayer sdk.AccAddress) channeltypesv2.RecvPacketResult
-	OnTimeoutPacket func(ctx context.Context, sourceID string, destinationID string, data channeltypesv2.PacketData, relayer sdk.AccAddress) error
+	OnSendPacket            func(ctx context.Context, sourceID string, destinationID string, sequence uint64, data channeltypesv2.PacketData, signer sdk.AccAddress) error
+	OnRecvPacket            func(ctx context.Context, sourceID string, destinationID string, data channeltypesv2.PacketData, relayer sdk.AccAddress) channeltypesv2.RecvPacketResult
+	OnTimeoutPacket         func(ctx context.Context, sourceID string, destinationID string, data channeltypesv2.PacketData, relayer sdk.AccAddress) error
+	OnAcknowledgementPacket func(ctx context.Context, sourceID string, destinationID string, data channeltypesv2.PacketData, acknowldgement []byte, relayer sdk.AccAddress) error
```
Comment (bznein) [inline testing/mock/v2/ibc_app.go:15]: we could probably rename all `sourceID` and `destinationID` here but it can be done in a separate PR imho

Comment (bznein) [reply testing/mock/v2/ibc_app.go]: Ah I think actually Cian did it in his PR. I will take care of merge conflicts

**Review by damiannolan — APPROVED** [review]
lgtm

File: modules/core/api/module.go [hunk 2]
```diff
@@ -41,4 +39,14 @@ type IBCModule interface {
 		data channeltypesv2.PacketData,
 		relayer sdk.AccAddress,
 	) error
+
+	// OnAcknowledgementPacket is executed when a packet gets acknowledged
+	OnAcknowledgementPacket(
+		ctx context.Context,
+		sourceID string,
+		destinationID string,
```
Comment (damiannolan) [inline modules/core/api/module.go:47]: should be sourceChannel / destinationChannel

File: testing/mock/v2/ibc_app.go [hunk 3]
```diff
@@ -9,7 +9,8 @@ import (
 )
 
 type IBCApp struct {
-	OnSendPacket    func(ctx context.Context, sourceID string, destinationID string, sequence uint64, data channeltypesv2.PacketData, signer sdk.AccAddress) error
-	OnRecvPacket    func(ctx context.Context, sourceID string, destinationID string, data channeltypesv2.PacketData, relayer sdk.AccAddress) channeltypesv2.RecvPacketResult
-	OnTimeoutPacket func(ctx context.Context, sourceID string, destinationID string, data channeltypesv2.PacketData, relayer sdk.AccAddress) error
+	OnSendPacket            func(ctx context.Context, sourceID string, destinationID string, sequence uint64, data channeltypesv2.PacketData, signer sdk.AccAddress) error
+	OnRecvPacket            func(ctx context.Context, sourceID string, destinationID string, data channeltypesv2.PacketData, relayer sdk.AccAddress) channeltypesv2.RecvPacketResult
+	OnTimeoutPacket         func(ctx context.Context, sourceID string, destinationID string, data channeltypesv2.PacketData, relayer sdk.AccAddress) error
+	OnAcknowledgementPacket func(ctx context.Context, sourceID string, destinationID string, data channeltypesv2.PacketData, acknowledgement []byte, relayer sdk.AccAddress) error
```
Comment (damiannolan) [inline testing/mock/v2/ibc_app.go:15]: ditto

File: testing/mock/v2/ibc_module.go [hunk 4]
```diff
@@ -49,14 +49,12 @@ func (im IBCModule) OnRecvPacket(ctx context.Context, sourceID string, destinati
...
 
-//
-// func (im IBCModule) OnAcknowledgementPacket() error {
-//	if im.IBCApp.OnAcknowledgementPacket != nil {
-//		return im.IBCApp.OnAcknowledgementPacket(...)
-//	}
-//	return nil
-// }
-//
+func (im IBCModule) OnAcknowledgementPacket(ctx context.Context, sourceID string, destinationID string, data channeltypesv2.PacketData, acknowledgement []byte, relayer sdk.AccAddress) error {
```
Comment (damiannolan) [inline testing/mock/v2/ibc_module.go:52]: ditto!

**Review by damiannolan — CHANGES_REQUESTED** [review]
whoops

**Pull request merged**